varinfo/cmr_search.py

```diff
@@ -134,3 +143,41 @@ def download_granule(granule_link: str,
         # Custom exception for error from `requests.get`
         raise GranuleDownloadException(
             str(requests_exception)) from requests_exception
+
+
+def get_edl_token_from_launchpad(launchpad_token: str,
+                                 cmr_env: CmrEnvType) -> Union[str, None]:
+    ''' Retrieve an EDL token given a LaunchPad token.
+        * launchpad_token: A LaunchPad token with no header prefixes:
+            <Launchpad token>
+        * cmr_env/mode: CMR environments (OPS, UAT, and SIT)
+    '''
+    url_urs_endpoint = UrsEnvsEdlTokenEndpoints.get(cmr_env)
+    try:
+        response = requests.post(url=url_urs_endpoint,
+                                 data=f'token={launchpad_token}',
+                                 timeout=10)
+        response.raise_for_status()
+
+    except Exception as requests_exception:
+        raise GetEdlTokenException(
+            str(requests_exception)) from requests_exception
+
+    return response.json().get('access_token')
+
+
+def get_edl_token_header(auth_header: str, cmr_env: CmrEnvType) -> str:
+    ''' Helper function for getting the header for an EDL token.
+        * auth_header: Authorization HTTP header, either:
+            - A header with a LaunchPad token: 'Authorization: <token>'
+            - An header with an EDL bearer token:
+                'Authorization: Bearer <token>'
+        * cmr_env/mode: CMR environments (OPS, UAT, and SIT)
+    '''
+    if 'Bearer ' not in auth_header:
+        edl_auth_header = (
+            f'Bearer {get_edl_token_from_launchpad(auth_header, cmr_env)}'
+        )
+    else:
+        edl_auth_header = auth_header
+    return edl_auth_header
```

Comment:
Nice - this makes a lot of sense. And I like that you split out the `get_edl_token_from_launchpad` to keep this function at a consistent level of abstraction. (Very "clean code"-esque)